src/main/java/net/minecraftforge/common/extensions/IForgeEntity.java

```diff
@@ -88,6 +89,9 @@ default ItemStack getPickedResult(HitResult target)
         ItemStack result = self().getPickResult();
         if (result == null) {
             SpawnEggItem egg = SpawnEggItem.byId(self().getType());
+            if (egg == null) {
+                egg = ForgeSpawnEggItem.fromEntityType(self().getType());
+            }
```

Comment:
Personally I think this should be patched into `SpawnEggItem.byId` so that mods that call that method continue to function.

Reply:
Lex wanted this to be done with the least possible amount of patches which is why I did it this way. In terms of only needing to call one method instead of two to cover all eggs, I suggested calling `SpawnEggItem.byId()` from `ForgeSpawnEggItem.fromEntityType()` on Discord in #issues-and-prs.

Reply:
Then we should at least have a utility method somewhere that does this and SpawnEggItem.byId should be deprecated - because it is not accurate.

Reply:
I would argue that `ForgeSpawnEggItem.fromEntityType()` is that utility method when the additional call to `SpawnEggItem.byId()` is added to it.

Reply:
Add a utility method that does calls vanilla's byId and falls back on Forge's method if that is not empty, so we have a "entity type -> spawn egg" conversion. You can use this method from here.